test/IECore/PointsAlgoTest.py

```diff
@@ -283,6 +283,143 @@ def testPrimitiveVariablesCorrectlyFiltered(self):
 		self.assertEqual( points["e"].data, IECore.FloatVectorData( range( 0, 10, 2 ) ) )
 		self.assertEqual( points["e"].interpolation, IECore.PrimitiveVariable.Interpolation.FaceVarying)
 
+class MergePointsTest( unittest.TestCase ) :
+
+	#todo find a better place for this decorator to live
+	#decorating a testfunction gives me time & pid to connect gdb to look at test failures.
+	def debugger(fn):
+
+		def wrapper(*args, **kwargs):
+			import os
+			import time
+			import sys
+			print "pid {0} ".format( os.getpid() )
+			sleeptime = 15
+			print "sleeping for {0}s".format(sleeptime)
+			time.sleep(sleeptime)
+
+			return fn(*args, **kwargs)
+
+		return wrapper
+
+	def testCanMergeTwoPointsPrimitivesWithNoPrimvars( self ) :
+
+		pointsA = IECore.PointsPrimitive( IECore.V3fVectorData( [ IECore.V3f( x ) for x in range( 0, 3 ) ] ) )
+		pointsB = IECore.PointsPrimitive( IECore.V3fVectorData( [ IECore.V3f( x ) for x in range( 5, 9) ] ) )
+
+		mergedPoints = IECore.PointsAlgo.mergePoints( [pointsA, pointsB] )
+
+		self.assertEqual( mergedPoints.numPoints, 3 + 4 )
+
+		self.assertEqual( mergedPoints["P"].data[0], IECore.V3f(0)	)
```

Comment:
some odd spacing here... i resisted mentioning spacing in the cpp, though there is a bit that's off there too. maybe give the whole thing a run through the clion claw+L?